modules/test/playwright/fixtures/clientExtensionsPageTest.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {test} from '@playwright/test';
+
+import {ClientExtensionsPage} from '../pages/product-navigation-applications-menu/ClientExtensionsPage';
+
+const clientExtensionsPageTest = test.extend<{
+	clientExtensionsPage: ClientExtensionsPage;
+}>({
+	clientExtensionsPage: async ({page}, use) => {
+		const clientExtensionsPage = new ClientExtensionsPage(page);
+
+		await clientExtensionsPage.goto();
```

Comment:
My intent here is to set page as a fixture, so that all tests in `editorConfigContributor.spec.ts` run on the Application Menu > Client Extensions page. I am not sure if there is a cleaner way to achieve this.

Reply:
I'd say you can use beforeEach in the test itself, like it is used here -> https://playwright.dev/docs/test-timeouts#change-timeout-from-a-beforeeach-hook

The fixture is OK to inject the CX page, like we do with some other pages, but navigating to the page is usually done in the tests themselves, since it is a navigation on which the tests depend and the more explicit the process in the tests the better for the future.

If we hide too much in the fixtures it can be very difficult to trace what tests are really doing.

In fact, if a test depended on two fixtures providing two different pages, both would fight to go to their page, so this is clearly wrong, I think...

Reply:
But, what you are suggesting negates the whole point of page fixtures. See https://playwright.dev/docs/test-fixtures#creating-a-fixture, they are using it in a similar way as in this PR:
```js
export const test = base.extend<MyFixtures>({
  todoPage: async ({ page }, use) => {
    // Set up the fixture.
    const todoPage = new TodoPage(page);
    await todoPage.goto();

    // Use the fixture value in the test.
    await use(todoPage);
  },
});
```

This is not only about convenience in tests, but also performance. Navigation is an expensive action, adding couple of seconds to each test. It will quickly pile up.

Difficulty to trace is a valid concern, but we can somewhat offset it with naming conventions for page names (`clientExtensionsPage` in this case).

Reply:
Mmm, you are right in the sense that you can only have one starting page in a test, which would be provided by the fixture and would set your test in a defined starting place.

However, what if you then navigate to another page in that same test?  How's the canonical way of using that new page's POM (page object model)? Do you instantiate it directly in the test? Encapsulate it inside the home page of the test as Miguel was doing in his PR? ... :thinking:

Reply:
OK, reading your link now I see that they do the navigation in the test file itself, even though they do it inside a composable fixture that provides a todoPage and a settingsPage.

So my point is still valid, I think. I'm requesting a separation of concerns in this case bc I assume that everything outside of tests (in this case your fixture inside the `fixtures` directory) can be reused by any test anywhere, thus it should be effect free.

So it's a matter of how we organize code in our scenario, not a problem of navigating inside a fixture which, as you said, it's a valid action if done in the proper place.

Reply:
Fixtures are run with each test, repeating the navigation. To confirm this, I added to the end of first test:
```js
await clientExtensionsPage.page.goto('/o/api');
```
![image](https://github.com/liferay-frontend/liferay-portal/assets/5435511/0163ce98-b2fe-401c-9258-f8d70cfd6629)

The second one completed successfully, starting test from home page:
![image](https://github.com/liferay-frontend/liferay-portal/assets/5435511/5aef99eb-76b4-4c04-9a8e-72713285aa38)


What's the difference between this and putting navigation in each test, or `beforeEach`?

Reply:
> OK, reading your link now I see that they do the navigation in the test file itself, even though they do it inside a composable fixture that provides a todoPage and a settingsPage.

What do you mean? :thinking: I don't see a difference between the sample and PR, both navigation are in fixture definition, in `test.extend`, in separate files from tests. The only difference is they also add reference to another page, `settingsPage`.

Reply:
> What's the difference between this and putting navigation in each test, or beforeEach?

A conceptual difference, even though both setups work.

If you put the navigation in the fixture and then inject the fixture in the test and don't navigate in the test you are assuming that the fixture contract is "when I enter the test execution the test browser will be in a certain page because of the fixture".

But now you can create another fixture that injects another page and navigates to another page, and then make your test depend on it too and in that case your test + the contracts from the two fixtures will be inconsistent, because: the test won't navigate anywhere, fixture A will "guarantee" it puts you in page X, fixture B  will "guarantee" that it puts you in Y page.

Reply:
> But now you can create another fixture that injects another page and navigates to another page, and then make your test depend on it too and in that case your test + the contracts from the two fixtures will be inconsistent, because: the test won't navigate anywhere, fixture A will "guarantee" it puts you in page X, fixture B will "guarantee" that it puts you in Y page.

Yeah, that's a good point! I guess my mental model was of a unique page fixture, that makes no sense to mix with other fixtures. But if we want to make it reusable by independent sets of tests, as a way to expose page-specific locators etc, then internal navigation would clash.

I'll make the change.